Fix various trackpads, mice and keyboards being recognized as gamepads on Linux/*BSD

Calinou opened the pull request:
This relies on the reported device name to exclude the devices based on substrings contained in the name.

- This closes https://github.com/godotengine/godot/issues/110906.
- See https://github.com/godotengine/godot/issues/59250#issuecomment-2560422205.

Thanks @TheBoctor and @geekley for the investigation 🙂

Calinou commented:
The file modified by this PR was removed entirely as part of the move to SDL, so I guess this poses the question:

@Nintorch Any ideas of whether we should still attempt this fix, or just try to do it upstream?

Nintorch commented:
While we're at it, maybe we also propose adding a `banned_words` list that we had before to SDL :D

Calinou commented:
> I think we can fix the issue in Godot first and after that we can propose the change to upstream SDL.

What would be the best location to attempt this in Godot code now?

Calinou committed:
Fix various trackpads, mice and keyboards being recognized as gamepads on Linux/*BSD

This relies on the reported device name to exclude the devices based on substrings
contained in the name.

Calinou commented:
@GameBeastProductions Can you [test](https://docs.godotengine.org/en/stable/contributing/workflow/testing_pull_requests.html) this PR locally to see if it resolves the issue on your end?

Nintorch commented on thirdparty/sdl/joystick/SDL_gamepad.c:
```diff
@@ -2676,6 +2676,19 @@ bool SDL_ShouldIgnoreGamepad(Uint16 vendor_id, Uint16 product_id, Uint16 version
         // Don't treat the Wii extension controls as a separate gamepad
         return true;
     }
+    if (SDL_strstr("TouchPad", name) || // Matches e.g. "SynPS/2 Synaptics TouchPad"
```
I think the arguments should be swapped 😅 (See line 2668, `SDL_strstr(name, " IMU")`)

Nintorch commented on thirdparty/sdl/joystick/SDL_gamepad.c:
```diff
@@ -2676,6 +2676,19 @@ bool SDL_ShouldIgnoreGamepad(Uint16 vendor_id, Uint16 product_id, Uint16 version
+        SDL_strstr("Mouse", name) || // Matches e.g. "Mouse passthrough"
+        SDL_strstr("Pen", name) || // Matches e.g. "Wacom One by Wacom S Pen"
+        SDL_strstr("Finger", name) || // Matches e.g. "Wacom HID 495F Finger"
+        SDL_strstr("Led", name)) { // Matches e.g. "ASRock LED Controller"
```
```suggestion
        SDL_strstr(name, "LED")) { // Matches e.g. "ASRock LED Controller"
```
Should it be all uppercase because "LED" is uppercase in the device's name?

Calinou replied:
Yes, the check is case-sensitive. Fixed.

Nintorch commented:
Can I borrow this list of banned words and make a PR for SDL with them? :D

Nintorch commented:
I created a PR in SDL's repository with my (slightly different) approach of solving this problem
